test/e2e/user_quota_tracing/user_quota_tracing_test.go

```diff
@@ -0,0 +1,261 @@
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package user_quota_tracing_test
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/apache/yunikorn-core/pkg/common/configs"
+	tests "github.com/apache/yunikorn-k8shim/test/e2e"
+	"github.com/apache/yunikorn-k8shim/test/e2e/framework/helpers/common"
+	"github.com/apache/yunikorn-k8shim/test/e2e/framework/helpers/k8s"
+	"github.com/apache/yunikorn-k8shim/test/e2e/framework/helpers/yunikorn"
+	siCommon "github.com/apache/yunikorn-scheduler-interface/lib/go/common"
+)
+
+const (
+	NANESPACE_LENGTH  = 10
+	WAIT_INTERVAL     = 60
+	DEFAULT_PARTITION = "default"
+)
+
+var _ = Describe("QuotaTracking: Two leaf queus for two groups", func() {
+	var ns string
+	BeforeEach(func() {
+		ns = "ns-" + common.RandSeq(NANESPACE_LENGTH)
+		By(fmt.Sprintf("Create namespace: %s", ns))
+		var ns1, err1 = kClient.CreateNamespace(ns, nil)
+		Ω(err1).NotTo(HaveOccurred())
+		Ω(ns1.Status.Phase).To(Equal(v1.NamespaceActive))
+	})
+
+	It("User qauta trace with 3 users and 2 groups", func() {
+		groups := []string{"group1", "group2"}
+		users := []string{"user1", "user2", "user3"}
+		queuePath := []string{"root", "root.group1_resources", "root.group2_resources"}
+		zeroPod := map[string]int64{siCommon.CPU: 0, siCommon.Memory: 0}
+		onePod := map[string]int64{siCommon.CPU: 100, siCommon.Memory: 50}
+		twoPods := map[string]int64{siCommon.CPU: 200, siCommon.Memory: 100}
+		yunikorn.UpdateCustomConfigMapWrapper(oldConfigMap, "", annotation, func(sc *configs.SchedulerConfig) error {
+			// remove placement rules so we can control queue
+			sc.Partitions[0].PlacementRules = nil
+			queuesConfigs := []struct {
+				partition, parentQueue, QueueName string
+			}{
+				{DEFAULT_PARTITION, queuePath[0], "group1_resources"},
+				{DEFAULT_PARTITION, queuePath[0], "group2_resources"},
+			}
+			for _, queueConfig := range queuesConfigs {
+				By(fmt.Sprintf("Add child queue %s to the parent queue %s", queueConfig.QueueName, queueConfig.parentQueue))
+				config := configs.QueueConfig{
+					Name: queueConfig.QueueName,
+				}
+				if err := common.AddQueue(sc, queueConfig.partition, queueConfig.parentQueue, config); err != nil {
+					return err
+				}
+			}
```

Comment:
I'm not overly familiar with this helper method, but can't we just replace the queue config inside the `SchedulerConfig` object with sth which initialized before? It's very hard to read even for something as simple as this.